core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -239,4 +254,33 @@ public boolean waitForReady(int timeoutMs) {
       return false;
     }
   }
+
+  @Override
+  protected void startCommonServices() {
+    boolean isStandby = mLeaderSelector.getState() == State.STANDBY;
+    if (ServerConfiguration.getBoolean(
+        PropertyKey.STANDBY_MASTER_METRICS_SINK_ENABLED) == isStandby) {
+      LOG.info("Start metric sinks.");
+      MetricsSystem.startSinks(
+          ServerConfiguration.get(PropertyKey.METRICS_CONF_FILE));
+    }
+    if (ServerConfiguration.getBoolean(
+        PropertyKey.STANDBY_MASTER_WEB_ENABLED) == isStandby) {
+      LOG.info("Start web server.");
+      startServingWebServer();
+    }
+  }
+
+  @Override
+  protected void stopCommonServices() throws Exception {
+    boolean isStandby = mLeaderSelector.getState() == State.STANDBY;
```

Comment:
same here

Reply:
As the `stopCommonServices` here is called by `losePrimacy` from standby master only, so I can simplify this logic to:

If we configured `STANDBY_MASTER_METRICS_SINK_ENABLED` as `false`, we should stop the service when `losePrimacy`, otherwise, we should do nothing.